portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -58,8 +60,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         commentList.add(comment);
     }
 
+    if (numOfComments > commentList.size()) numOfComments = commentList.size();
```

Comment:
This line and the `subList` call below should be unnecessary. Assuming the `asIterable` call above properly returns no more than `numOfComments` number of comments, then `commentList` has the right number of comments. If, in the case you don't trust `asIterable` to return the right number of comments with the limit you have imposed, i.e. `commentList.size() > numOfComments`, then this if check is testing the incorrect condition. It should check if `commentList.size() > numOfComments` and then restrict the list of comments to the unmodified `numOfComments`.

Reply:
I'll add these changes in the next PR, I'm adding a little bit of visual enhancements to the site.